packages/flutter/lib/src/material/filled_button.dart

```diff
@@ -0,0 +1,754 @@
+  ///
+  /// * `padding`
+  ///   * `textScaleFactor <= 1` - start(12) end(16)
+  ///   * `1 < textScaleFactor <= 2` - lerp(start(12) end(16), horizontal(8))
+  ///   * `2 < textScaleFactor <= 3` - lerp(horizontal(8), horizontal(4))
+  ///   * `3 < textScaleFactor` - horizontal(4)
+  ///
+  /// The default value for `side`, which defines the appearance of the button's
+  /// outline, is null. That means that the outline is defined by the button
+  /// shape's [OutlinedBorder.side]. Typically the default value of an
+  /// [OutlinedBorder]'s side is [BorderSide.none], so an outline is not drawn.
+  ///
+  @override
+  ButtonStyle defaultStyleOf(BuildContext context) {
+    switch (_variant) {
+      case _FilledButtonVariant.filled:
+        return _FilledButtonDefaultsM3(context);
+      case _FilledButtonVariant.tonal:
+        return _FilledTonalButtonDefaultsM3(context);
+    }
+  }
+
+  /// Returns the [FilledButtonThemeData.style] of the closest
+  /// [FilledButtonTheme] ancestor.
+  @override
+  ButtonStyle? themeStyleOf(BuildContext context) {
+    return FilledButtonTheme.of(context).style;
+  }
+}
+
+EdgeInsetsGeometry _scaledPadding(BuildContext context) {
+  return ButtonStyleButton.scaledPadding(
+    const EdgeInsets.symmetric(horizontal: 16),
+    const EdgeInsets.symmetric(horizontal: 8),
+    const EdgeInsets.symmetric(horizontal: 4),
+    MediaQuery.maybeOf(context)?.textScaleFactor ?? 1,
+  );
+}
+
+@immutable
+class _FilledButtonDefaultColor extends MaterialStateProperty<Color?> with Diagnosticable {
+  _FilledButtonDefaultColor(this.color, this.disabled);
+
+  final Color? color;
+  final Color? disabled;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return disabled;
+    }
+    return color;
+  }
+}
+
+@immutable
+class _FilledButtonDefaultOverlay extends MaterialStateProperty<Color?> with Diagnosticable {
+  _FilledButtonDefaultOverlay(this.overlay);
+
+  final Color overlay;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.hovered)) {
+      return overlay.withOpacity(0.08);
+    }
+    if (states.contains(MaterialState.focused) || states.contains(MaterialState.pressed)) {
+      return overlay.withOpacity(0.24);
+    }
+    return null;
+  }
+}
+
+@immutable
+class _FilledButtonDefaultElevation extends MaterialStateProperty<double> with Diagnosticable {
+  _FilledButtonDefaultElevation(this.elevation);
+
+  final double elevation;
+
+  @override
+  double resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return 0;
+    }
+    if (states.contains(MaterialState.hovered)) {
+      return elevation + 2;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return elevation + 2;
+    }
+    if (states.contains(MaterialState.pressed)) {
+      return elevation + 6;
```

Comment:
I think this is available in tokens, but not sure

Reply:
Yup this was a left over from some backwards compatibility stuff from another button code that was copied. Fixed.